admob/integration_test/src/integration_test.cc

```diff
@@ -536,6 +553,237 @@ TEST_F(FirebaseAdMobTest, TestRequestConfigurationSetGet) {
+  SKIP_TEST_ON_DESKTOP;
+
+  firebase::admob::InterstitialAd* interstitial =
+      new firebase::admob::InterstitialAd();
+
+  WaitForCompletion(interstitial->Initialize(app_framework::GetWindowContext()),
+                    "Initialize");
+
+  TestFullScreenContentListener full_screen_content_listener;
+  interstitial->SetFullScreenContentListener(&full_screen_content_listener);
+
+  TestPaidEventListener paid_event_listener;
+  interstitial->SetPaidEventListener(&paid_event_listener);
+
+  // When the InterstitialAd is initialized, load an ad.
+  firebase::admob::AdRequest request = GetAdRequest();
+  WaitForCompletion(interstitial->LoadAd(kInterstitialAdUnit, request),
+                    "LoadAd");
+
+  WaitForCompletion(interstitial->Show(), "Show");
+
+  LogDebug("Click the Ad, and then return to the app to continue");
+
+  while (
+      full_screen_content_listener.num_on_ad_dismissed_full_screen_content_ ==
+      0) {
+    app_framework::ProcessEvents(1000);
+  }
+
+  LogDebug("Waiting for a moment to ensure all callbacks are recorded.");
+  app_framework::ProcessEvents(2000);
+
+  EXPECT_EQ(full_screen_content_listener.num_on_ad_clicked_, 1);
+  EXPECT_EQ(full_screen_content_listener.num_on_ad_showed_full_screen_content_,
+            1);
+  EXPECT_EQ(full_screen_content_listener.num_on_ad_impression_, 1);
+  EXPECT_EQ(paid_event_listener.num_on_paid_event_, 1);
+  EXPECT_EQ(
+      full_screen_content_listener.num_on_ad_dismissed_full_screen_content_, 1);
+
+  interstitial->SetFullScreenContentListener(nullptr);
+  interstitial->SetPaidEventListener(nullptr);
+
+  delete interstitial;
+}
+
+TEST_F(FirebaseAdMobTest, TestRewardedAdLoadAndShow) {
+  TEST_REQUIRES_USER_INTERACTION;
+  SKIP_TEST_ON_DESKTOP;
+  SKIP_TEST_ON_IOS;
+
+  firebase::admob::RewardedAd* rewarded = new firebase::admob::RewardedAd();
+
+  WaitForCompletion(rewarded->Initialize(app_framework::GetWindowContext()),
+                    "Initialize");
+
+  TestFullScreenContentListener full_screen_content_listener;
+  rewarded->SetFullScreenContentListener(&full_screen_content_listener);
+
+  TestPaidEventListener paid_event_listener;
+  rewarded->SetPaidEventListener(&paid_event_listener);
+
+  // When the RewardedAd is initialized, load an ad.
+  firebase::admob::AdRequest request = GetAdRequest();
+  WaitForCompletion(rewarded->LoadAd(kRewardedAdUnit, request), "LoadAd");
+
+  firebase::admob::RewardedAd::ServerSideVerificationOptions options;
+  // We cannot programmatically verify that the AdMob phone SDKs marshal
+  // these values properly (there are no get methods). At least invoke the
+  // method to ensure least we can set them without any exceptions occurring.
+  options.custom_data = "custom data";
+  options.user_id = "123456";
+  rewarded->SetServerSideVerificationOptions(options);
+
+  TestUserEarnedRewardListener user_earned_reward_listener;
+  WaitForCompletion(rewarded->Show(&user_earned_reward_listener), "Show");
+
+  LogDebug(
```

Comment:
I think you get credit for the ad even if you just watch it through, you don't have to install the thing.

Reply:
While you get the reward the ad doesn't produce the ad clicked event that we're also looking for.

Reply:
Done.